statsmodels/discrete/discrete_model.py

```diff
@@ -2371,11 +2371,10 @@ def __init__(self, model, mlefit, cov_type='nonrobust', cov_kwds=None,
 
 
     def __getstate__(self):
-        try:
-            #remove unpicklable callback
-            self.mle_settings['callback'] = None
-        except (AttributeError, KeyError):
-            pass
+        # remove unpicklable methods
+        self.mle_settings['callback'] = None
```

Comment:
I'm not clear on why there was a try/except in the first place.  `KeyError` shouldn't occur (maybe confusion with `__setstate__`?).  `AttributeError` would make sense if there is no `mle_settings` I guess, but I don't think that case is relevant is it?